worlds/tunic/__init__.py

```diff
@@ -497,24 +499,80 @@ def stage_pre_fill(cls, multiworld: MultiWorld) -> None:
+            out_of_spec_worlds = set()
 
             for filler_item in grass_fill:
-                grass_fill_locations.pop().place_locked_item(filler_item)
+                loc_to_fill = grass_fill_locations.pop()
+                try:
+                    loc_to_fill.place_locked_item(filler_item)
+                except Exception:
+                    out_of_spec_worlds.add(multiworld.worlds[loc_to_fill.item.player].game)
+                    for loc in backup_grass_locations:
+                        if not loc.item:
+                            loc.place_locked_item(filler_item)
+                            break
+                        else:
+                            out_of_spec_worlds.add(multiworld.worlds[loc_to_fill.item.player].game)
+                    else:
+                        raise Exception("TUNIC: Could not fulfill local_filler option. This issue is caused by another "
+                                        "world filling TUNIC locations during pre_fill.\n"
+                                        "Archipelago does not allow us to place items into the item pool after "
+                                        "create_items, so we cannot recover from this issue.\n"
+                                        f"This is likely caused by the following world(s): {out_of_spec_worlds}.\n"
+                                        f"Please let the world dev(s) for the listed world(s) know that there is an "
+                                        f"issue there.\n"
+                                        "As a workaround, you can try setting the local_filler option lower for "
+                                        "TUNIC slots with Breakable Shuffle or Grass Rando enabled. You may be able to "
+                                        "try generating again, as it may not happen every generation.")
 
             for filler_item in non_grass_fill:
-                non_grass_fill_locations.pop().place_locked_item(filler_item)
+                loc_to_fill = non_grass_fill_locations.pop()
+                try:
+                    loc_to_fill.place_locked_item(filler_item)
+                except Exception:
+                    out_of_spec_worlds.add(multiworld.worlds[loc_to_fill.item.player].game)
+                    for loc in backup_non_grass_locations:
+                        if not loc.item:
+                            loc.place_locked_item(filler_item)
+                            break
+                        else:
+                            out_of_spec_worlds.add(multiworld.worlds[loc_to_fill.item.player].game)
+                    else:
+                        raise Exception("TUNIC: Could not fulfill local_filler option. This issue is caused by another "
+                                        "world filling TUNIC locations during pre_fill.\n"
+                                        "Archipelago does not allow us to place items into the item pool after "
```

Comment:
Same question here.

Reply:
It's the same text, resolving this one, we can chat on the other one